src/misc/useOffset.ts

```diff
@@ -3,7 +3,7 @@ import useSearchParams from './useSearchParams';
 export const limit = 50;
 export default function useOffset(location: Location, keyName = 'offset') {
   const [searchParams, setSearchParams] = useSearchParams(location);
-  const offset = keyName in searchParams ? searchParams[keyName] : 0;
+  const offset = keyName in searchParams ? Number(searchParams[keyName]) : 0;
```

Comment:
As offset and limit should be integral, [`parseInt()` is better than `Number()`][1] here IMHO:

```suggestion
  const offset = keyName in searchParams ? parseInt(searchParams[keyName]) : 0;
```

[1]: https://stackoverflow.com/a/4090577/383405